rust/kcl-lib/src/std/mirror.rs

```diff
@@ -70,56 +65,81 @@ async fn inner_mirror_2d(
                 .await?;
-        }
-    };
 
-    // After the mirror, get the first child uuid for the path.
-    // The "get extrusion face info" API call requires *any* edge on the sketch being extruded.
-    // But if you mirror2d a sketch these IDs might change so we need to get the children versus
-    // using the IDs we already have.
-    // We only do this with mirrors because otherwise it is a waste of a websocket call.
-    for sketch in &mut starting_sketches {
-        let response = exec_state
-            .send_modeling_cmd(
-                (&args).into(),
-                ModelingCmd::from(mcmd::EntityGetAllChildUuids { entity_id: sketch.id }),
-            )
-            .await?;
-        let OkWebSocketResponseData::Modeling {
-            modeling_response:
-                OkModelingCmdResponse::EntityGetAllChildUuids(EntityGetAllChildUuids { entity_ids: child_ids }),
-        } = response
-        else {
-            return Err(KclError::new_internal(KclErrorDetails::new(
-                "Expected a successful response from EntityGetAllChildUuids".to_string(),
-                vec![args.source_range],
-            )));
-        };
+            if let OkWebSocketResponseData::Modeling {
+                modeling_response: OkModelingCmdResponse::EntityMirrorAcrossEdge(mirror_info),
+            } = &resp
+            {
+                let face_edge_info = &mirror_info.entity_face_edge_ids;
 
-        if child_ids.len() >= 2 {
-            // The first child is the original sketch, the second is the mirrored sketch.
-            let child_id = child_ids[1];
-            sketch.mirror = Some(child_id);
-        } else {
-            return Err(KclError::new_type(KclErrorDetails::new(
-                "Expected child uuids to be >= 2".to_string(),
-                vec![args.source_range],
-            )));
+                starting_sketches
+                    .iter_mut()
+                    .zip(face_edge_info.iter())
+                    .try_for_each(|(sketch, info)| {
+                        sketch.id = info.object_id;
+                        let first_edge = info.edges.first().copied();
+                        match first_edge {
+                            Some(edge) => sketch.mirror = Some(edge),
+                            None => {
+                                return Err(KclError::new_engine(KclErrorDetails::new(
+                                    "No edges found in mirror info".to_string(),
+                                    vec![args.source_range],
+                                )))
+                            }
+                        }
+                        Ok(())
+                    })?;
+            } else {
+                return Err(KclError::new_engine(KclErrorDetails::new(
+                    format!("EntityMirrorAcrossEdge response was not as expected: {:?}", resp),
+                    vec![args.source_range],
+                )));
+            };
         }
-    }
+    };
```

Comment:
```suggestion
    }
```

You need the semicolon when it's `let foo = match { ... };`. But there's no more `let` so I don't think we don't need it.